dsp/helpers/tables.c

```diff
@@ -103,3 +103,33 @@ const uint16_t * get_sc16q11_mag_12bit_table()
     return table;
 }
 
+// The magic scaler is (32767.0 / 2047.0)
+#define MAGIC_SCALER 32.01514f
+#define CONVERT_AND_SCALE(__in) \
+({ \
+    uint16_t __out = abs(le16toh(__in) - 2048); \
+    ceil(__out * 32.01514f); \
+})
```

Comment:
See comments in the impl file about this.

Given that it's only used in one place in this file, I'd just expand the expression and not bother with the macros.

Reply:
done